commit 5990378b393429244559f4750f2ee3a50929b932
Author: Felipe Balbi <ba...@ti.com>
Date:   Fri Feb 11 10:00:02 2011 +0200

    usb: musb: fix build breakage
    
    commit 0662481855c389b75a0a54c32870cc90563d80a9
    (usb: musb: disable double buffering when it's broken),
    introduced a compile error when gadget API is disabled.
    
    Fix it.
    
    Signed-off-by: Felipe Balbi <ba...@ti.com>

diff -Naur linux-2.6.38-rc5//drivers/usb/musb/musb_core.h linux-2.6.38-rc5a/drivers/usb/musb/musb_core.h
--- linux-2.6.38-rc5//drivers/usb/musb/musb_core.h	2011-02-16 04:23:45.000000000 +0100
+++ linux-2.6.38-rc5a/drivers/usb/musb/musb_core.h	2011-02-21 15:23:48.398271718 +0100
@@ -488,6 +488,15 @@
 	unsigned		set_address:1;
 	unsigned		test_mode:1;
 	unsigned		softconnect:1;
+
+	u8			address;
+	u8			test_mode_nr;
+	u16			ackpend;		/* ep0 */
+	enum musb_g_ep0_state	ep0_state;
+	struct usb_gadget	g;			/* the gadget */
+	struct usb_gadget_driver *gadget_driver;	/* its driver */
+#endif
+
 	/*
 	 * FIXME: Remove this flag.
 	 *
@@ -501,14 +510,6 @@
 	 */
 	unsigned                double_buffer_not_ok:1 __deprecated;
 
-	u8			address;
-	u8			test_mode_nr;
-	u16			ackpend;		/* ep0 */
-	enum musb_g_ep0_state	ep0_state;
-	struct usb_gadget	g;			/* the gadget */
-	struct usb_gadget_driver *gadget_driver;	/* its driver */
-#endif
-
 	struct musb_hdrc_config	*config;
 
 #ifdef MUSB_CONFIG_PROC_FS
